be/src/exec/analytor.cpp

```diff
@@ -758,6 +766,32 @@ int64_t Analytor::_find_first_not_equal(Column* column, int64_t target, int64_t
     return end - 1;
 }
 
+int64_t Analytor::_find_first_not_equal_for_hash_based_partition(int64_t target, int64_t start, int64_t end) {
+    // In this case, we cannot compare each column one by one like Analytor::_find_first_not_equal does,
+    // and we must compare all the partition columns for one comparation
+    auto compare = [this](size_t left, size_t right) {
+        for (auto& column : _partition_columns) {
+            auto res = column->compare_at(left, right, *column, 1);
+            if (res != 0) {
+                return res;
+            }
+        }
+        return 0;
+    };
+    while (start + 1 < end) {
```

Comment:
[start, end]'s elements >= [target] ?
usually the [start] may not equal, add a special judge for start ?
```
    if (compare(target, start) == 0) {
        return start;
    }
```